src/main/java/pl/danyboy/PasswordManager.java

```diff
@@ -1,23 +1,56 @@
 package pl.danyboy;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 public class PasswordManager {
-    private List<String> passwords;
-    private RandomGenerator randomGenerator = new RandomGenerator();
+
+    private final Map<String, Boolean> passwords = new HashMap<>();
```

Comment:
Tu jest jeden z tych bardzo rzadkich przypadków kiedy warto dodać komentarz. np. //true if password can be used, false is password was used during the game. Bo chwilę musiałem w kod popatrzeć, żeby ogarnąć kied false, kiedy true, a chyba nie ma ładnej, opisowe nazwy dal tej map :)